mitodl/mit-open
# header updates

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4243

### Description (What does it do?)
This PR updates the header to be as close to the current designs as possible, including both desktop and mobile layouts.

### Screenshots (if appropriate):
![image](https://github.com/mitodl/mit-open/assets/12089658/d4c1f4dd-cc29-4625-bd06-15f906a5d8e4)
![image](https://github.com/mitodl/mit-open/assets/12089658/d0659955-2947-4a88-9175-219f3afe9071)
![image](https://github.com/mitodl/mit-open/assets/12089658/dafa92f5-f7aa-413a-96be-c02eedfce8bf)
![image](https://github.com/mitodl/mit-open/assets/12089658/34c60409-b1bf-4006-b0ca-6638876b1d6f)
![image](https://github.com/mitodl/mit-open/assets/12089658/532806b0-226e-41fc-b0f6-9ee8001e239d)
![image](https://github.com/mitodl/mit-open/assets/12089658/6e31fb1c-1844-4438-9e26-b0e9d0a22989)

### How can this be tested?
 - Spin up your local `mit-open` and make sure you are set up for logging in through Keycloak, or you have a Django admin account that you can log into
 - For both mobile and desktop, verify that the header looks and functions as expected based on the Figma designs in the linked issue:
   - First view the home page (http://localhost:8063) while not logged in. 
   - Click the login button and log in (if you don't have Keycloak set up you can log in at http://localhost:8063/admin with your superuser, then return to the home page)
   - Click the search icon to go to the search page
   - Click the MIT icon in the upper left to return home
